dashboard/agent.py

```diff
@@ -361,36 +381,27 @@ async def _check_parent():
     except Exception as e:
         # All these env vars should be available because
         # they are provided by the parent raylet.
-        restart_count = os.environ["RESTART_COUNT"]
-        max_restart_count = os.environ["MAX_RESTART_COUNT"]
         raylet_pid = os.environ["RAY_RAYLET_PID"]
         node_ip = args.node_ip_address
-        if restart_count >= max_restart_count:
-            # Agent is failed to be started many times.
-            # Push an error to all drivers, so that users can know the
-            # impact of the issue.
-            gcs_publisher = GcsPublisher(args.gcs_address)
-            traceback_str = ray._private.utils.format_error_message(
-                traceback.format_exc()
-            )
-            message = (
-                f"(ip={node_ip}) "
-                f"The agent on node {platform.uname()[1]} failed to "
-                f"be restarted {max_restart_count} "
-                "times. There are 3 possible problems if you see this error."
-                "\n  1. The dashboard might not display correct "
-                "information on this node."
-                "\n  2. Metrics on this node won't be reported."
-                "\n  3. runtime_env APIs won't work."
-                "\nCheck out the `dashboard_agent.log` to see the "
-                "detailed failure messages."
-            )
-            ray._private.utils.publish_error_to_driver(
-                ray_constants.DASHBOARD_AGENT_DIED_ERROR,
-                message,
-                redis_client=None,
-                gcs_publisher=gcs_publisher,
-            )
-            logger.error(message)
+        # Agent is failed to be started many times.
+        # Push an error to all drivers, so that users can know the
+        # impact of the issue.
+        redis_client = None
+        gcs_publisher = GcsPublisher(address=args.gcs_address)
+
+        traceback_str = ray._private.utils.format_error_message(traceback.format_exc())
+        message = (
```

Comment:
Do we need it? In this case, raylet will just fail right?

Reply:
Yeah, if we open fate sharing by default, we don't need this. I can remove this.